pkg/skaffold/runner/runcontext/context.go

```diff
@@ -147,11 +158,12 @@ func (rc *RunContext) Deployers() []latest.DeployType { return rc.Pipelines.Depl
 func (rc *RunContext) TestCases() []*latest.TestCase { return rc.Pipelines.TestCases() }
 
 func (rc *RunContext) StatusCheck() bool {
-	sc := rc.Opts.StatusCheck.Value()
-	if sc == nil {
-		return true
+	scOpts := rc.Opts.StatusCheck.Value()
+	scPipelines := rc.Pipelines.StatusCheck()
+	if scOpts == nil {
+		return scPipelines
 	}
-	return *sc
+	return *scOpts
```

Comment:
this logic breaks current behavior?

eg: If I don't define `statusCheck` in my `skaffold.yaml` it'll be set to false. Since I don't set the flag either, it'll skip status check (line 164).

Reply:
it should rather be:

```go
	scOpts := rc.Opts.StatusCheck.Value()
	scConfig := rc.Pipelines.StatusCheck()
	switch {
	case scOpts != nil:
		return *scOpts
	case scConfig != nil:
		return *scConfig
	default:
		return true
	}
```

Reply:
👍  Updated! (My initial implementation had `skaffold fix` automatically add a `StatusCheck: true` stanza to the skaffold.yaml in order to not break the enabled-by-default behavior, but changing its type to `*bool` instead to distinguish false from unspecified is much better, thanks!)